src/compiler/compile_engine.cc

```diff
@@ -202,6 +202,24 @@ class CompileEngine {
       Array<Tensor> out = fcompute[inode.source->op()](
           inode.source->attrs, op_inputs, out_info);
       CHECK_EQ(out.size(), inode.source->num_outputs());
+
+      // check output dimentions also match
+      // This check is to make sure the NNVM operator Infer match with Compute result.
+      // Missing this check may pass the build but leads to runtime errors.
+      for (uint32_t i=0; i < out.size(); i++) {
+        CHECK_EQ(out[i].ndim(), out_info[i].ndim()) << inode.source->op()->name;
+        tvm::Tensor in_tensor = out[i];
```

Comment:
the name in_tensor is confusing here

Reply:
Updated with inferred_tensor, computed_tensor instead :)